spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducer.java

```diff
@@ -0,0 +1,228 @@
+	 * {@link #createConsumerGroup}. If not set, the defined bean name {@link #getBeanName()} is used.
+	 * @param consumerGroup the Consumer Group on which this adapter should register to listen messages.
+	 */
+	public void setConsumerGroup(@Nullable String consumerGroup) {
+		this.consumerGroup = consumerGroup;
+	}
+
+	/**
+	 * Set the name of the consumer. When a consumer name is provided, this adapter is switched to the Consumer Group
+	 * feature. Note that this value should be unique in the group
+	 * @param consumerName the consumer name in the Consumer Group
+	 */
+	public void setConsumerName(@Nullable String consumerName) {
+		this.consumerName = consumerName;
+	}
+
+	/**
+	 * Create the Consumer Group if and only if it does not exist. During the
+	 * creation we also create the stream, see {@code MKSTREAM}.
+	 * @param createConsumerGroup specify if we should create the Consumer Group, {@code false} by default
+	 */
+	public void setCreateConsumerGroup(boolean createConsumerGroup) {
+		this.createConsumerGroup = createConsumerGroup;
+	}
+
+	/**
+	 * Set {@link StreamOperations} used to customize the {@link StreamReceiver}.
+	 * It provides a way to set the polling timeout and the serialization context.
+	 * By default the polling timeout is set to infinite and {@link StringRedisSerializer} is used.
+	 * @param streamReceiverOptions the desired receiver options
+	 * */
+	public void setStreamReceiverOptions(@Nullable StreamReceiver.StreamReceiverOptions<String, ?> streamReceiverOptions) {
+		this.streamReceiverOptions = streamReceiverOptions;
+	}
+
+	@Override
+	public String getComponentType() {
+		return "redis:stream-inbound-channel-adapter";
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		this.streamReceiver = StreamReceiver.create(this.reactiveConnectionFactory, this.streamReceiverOptions);
+		if (StringUtils.hasText(this.consumerName) && StringUtils.isEmpty(this.consumerGroup)) {
+			this.consumerGroup = getBeanName();
+		}
+		this.reactiveRedisTemplate = new ReactiveRedisTemplate<>(this.reactiveConnectionFactory,
+				RedisSerializationContext.string());
+		this.reactiveStreamOperations = this.reactiveRedisTemplate.opsForStream();
+	}
+
+	@Override
+	protected void doStart() {
+		super.doStart();
+
+		StreamOffset<String> offset = StreamOffset.create(this.streamKey, this.readOffset);
+
+		Flux<Message<?>> messageFlux;
+
+		if (StringUtils.isEmpty(this.consumerName)) {
+			messageFlux = this.streamReceiver
+					.receive(offset)
+					.map(event -> getMessageBuilderFactory().withPayload(this.extractPayload ? event.getValue() : event)
+							.setHeader(RedisHeaders.STREAM_KEY, event.getStream())
+							.setHeader(RedisHeaders.STREAM_MESSAGE_ID, event.getId())
+							.build());
+		}
+		else {
+			Assert.notNull(this.consumerGroup, "'consumerGroup' must not be null");
+
+			if (this.createConsumerGroup) {
+				Assert.hasText(this.consumerName, "'consumerName' must be set");
```

Comment:
This is redundant: guarded by the previous `if (StringUtils.isEmpty(this.consumerName)) {`